### PR TITLE
refactor(loader-bar): refactor component to typescript - FE-4805

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour

- Refactor component to be TypeScript
- Refactor test story to be TypeScript

### Current behaviour

Component is currently JS and test story is MDX

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions

Made sure component behaves and looks the same as it did prior to this work.

### CODE REVIEW

File: src/components/loader-bar/loader-bar.component.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  return (
+    <StyledLoader {...rest} {...tagComponent("loader-bar", rest)}>
+      <StyledLoaderBar size={size}>
+        <InnerBar size={size} />
+      </StyledLoaderBar>
+    </StyledLoader>
+  );
+};
+
+LoaderBar.displayName = "LoaderBar";
```
Comment (mkrds) [inline src/components/loader-bar/loader-bar.component.tsx:25]: Do we need this line? It seems redundant

File: src/components/loader-bar/loader-bar.style.ts [hunk 2]
```diff
@@ -23,7 +37,7 @@ const innerBarAnimation = keyframes`
   }
 `;
 
-const StyledLoaderBar = styled.div`
+const StyledLoaderBar = styled.div<StyledLoaderBarProps>`
```
Comment (mkrds) [inline src/components/loader-bar/loader-bar.style.ts:40]: I does not look like StyledLoaderBar accepts any MarginProps

File: src/components/loader-bar/loader-bar.style.ts [hunk 3]
```diff
@@ -34,7 +48,7 @@ const StyledLoaderBar = styled.div`
   `}
 `;
 
-const InnerBar = styled.div`
+const InnerBar = styled.div<StyledLoaderBarProps>`
```
Comment (mkrds) [inline src/components/loader-bar/loader-bar.style.ts:51]: I does not look like InnerBar accepts any MarginProps

File: src/components/loader-bar/loader-bar.component.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  StyledLoader,
+  StyledLoaderBarProps,
+} from "./loader-bar.style";
+
+export interface LoaderBarProps extends StyledLoaderBarProps, MarginProps {}
+
+const LoaderBar = ({
+  size = "medium",
+  ...rest
+}: LoaderBarProps): JSX.Element => {
```
Comment (mkrds) [inline src/components/loader-bar/loader-bar.component.tsx:15]: It's not really necessary to state that `JSX.Element` is returned from a component as usually its inferred automatically by TS

Comment (DipperTheDan) [reply src/components/loader-bar/loader-bar.component.tsx:15]: I'll get it removed.

**Comment by Parsium** [comment]
Happy to approve once @mkrds's comments have been resolved 👍🏼

**Review by mkrds — DISMISSED** [review]

**Review by Parsium — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by Parsium — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**